### DISCUSSION TITLE
Embeddings dimensions

### DISCUSSION THREAD

**Discussion by carlos-aguayo** [pyannote/pyannote-audio#1512] [Q&A]
If I get the embeddings using speaker-diarization
```
p = Pipeline.from_pretrained("pyannote/speaker-diarization-3.0", use_auth_token=True)
diarization, embeddings = p(file, return_embeddings=True)
````
Then the dimension of embeddings is 256.

If I get the embeddings using:
```
model = Model.from_pretrained("pyannote/embedding", use_auth_token=True)
inference = Inference(model, window="whole")
embedding = inference.crop(file, segment)
```
Then the dimension of the embedding is 512

What can I do to ensure that I get the same dimension on both invocations?

**Comment by hbredin** [comment]
`pyannote/speaker-diarization-3.0` does [**not**](https://huggingface.co/pyannote/speaker-diarization-3.0/blob/main/config.yaml#L7) rely on `pyannote/embedding` internally.
Instead, it relies on [`hbredin/wespeaker-voxceleb-resnet34-LM`](https://github.com/pyannote/pyannote-audio/blob/4b09aefe4093a681b258626523a1408d754dd90c/pyannote/audio/pipelines/speaker_verification.py#L389-L412), hence the different dimension.
